packages/libs/react-ui/CONTRIBUTING.md

```diff
@@ -0,0 +1,214 @@
+# Contributing to @kadena/react-ui
+
+This is an aggregation of the conventions that we have picked up while
+developing this component library. This library is still in its early stages of
+development so please make any suggestions if you see ways it can be improved!
+
+## Component Conventions
+
+### Component Structure
+
+We have had discussions around whether we want to use composition or
+configuration as the pattern for developing components and have decided that
+**in most cases we want to use composition, but it differs on a case by case
+basis**. [Composition over Configuration][1]
+
+#### Composition - Subcomponent structure
+
+Composition makes sense if the component is more complex and needs a bit more
+flexibility in regards to data, children, or layout. Components using this this
+pattern will export one object of subcomponents that will make up the building
+blocks for the entire component. These subcomponents will always export one
+`Root` component as the containing component. This is inspired by [Radix
+Primitives][2].
+
+A simple example of a component that should use this pattern is the Grid
+component.
+
+```jsonc
+<Grid.Root>
+    <Grid.Item>{item1}</Grid.Item>
+    <Grid.Item>{item2}</Grid.Item>
+    <Grid.Item>{item3}</Grid.Item>
+</Grid.Root>
+```
+
+The reason we opted for this approach when it comes to more complex components
+is because this allows us to keep our components more flexible by providing
+styled building blocks that users can freely position based on their own unique
+logic. It allows us to keep complexity outside of our component logic so we
+don't need to make exceptions or combinations based on prop configuration.
+
+#### Configuration - Configuring via props
+
+This approach makes sense when developing simpler components. So in cases of any
+`atom` or `molecule`<sup>\*</sup> sized components (ex. Box, Text, Button),
+often times using the subcomponent approach isn't necessary because the
+different iterations of how that component will look are limited and introducing
+more subcomponents would negatively affect DX.
+
+> <sup>\*</sup> These terms come from the Atomic Design methodology. If you're
+> unfamiliar with this, you can read more about it
+> [here](https://bradfrost.com/blog/post/atomic-web-design/) and
+> [here](https://atomicdesign.bradfrost.com/chapter-2/).
+
+One example of a component that we updated from the Composition to Configuration
+approach was the `Button` component.
+
+Previously when the `Button` was used with an icon, it needed to be composed
+like this:
+
+```jsonc
```

Comment:
I thought md has support for `jsx`?

Reply:
I just copied it from the other file, but this is better! Updated 👍